### PR TITLE
Implement monster creation function

### PR DESCRIPTION
**Pull request by IHIHHIHI**
This revision includes:
- Add component 'Health'
- Handle monster health
- Add function 'CreateMonster()'
- Make monster with properties
- Make monster and check

### CODE REVIEW

File: Sources/CubbyTower/Helpers/MonsterHelpers.cpp [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+// personal capacity and are not conveying any rights to any intellectual
+// property of any third parties.
+
+#include <CubbyTower/Commons/Tags.hpp>
+#include <CubbyTower/Components/Distance.hpp>
+#include <CubbyTower/Components/Health.hpp>
+#include <CubbyTower/Components/Position.hpp>
+#include <CubbyTower/Components/TypeMask.hpp>
+#include <CubbyTower/Helpers/MonsterHelpers.hpp>
+#include <CubbyTower/Systems/PathSystem.hpp>
```
Comment (red1108) [inline Sources/CubbyTower/Helpers/MonsterHelpers.cpp]: `PathSystem.hpp` is not necessary

**Review by red1108 — CHANGES_REQUESTED** [review]
In MonsterHelpers.cpp, `PathSystem.hpp` is not necessary

File: Includes/CubbyTower/Helpers/MonsterHelpers.hpp [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+// property of any third parties.
+
+#ifndef CUBBYTOWER_MONSTER_HELPERS_HPP
+#define CUBBYTOWER_MONSTER_HELPERS_HPP
+
+#include <entt/entt.hpp>
+
+namespace CubbyTower
+{
+//! Creates a Monster
```
Comment (utilForever) [inline Includes/CubbyTower/Helpers/MonsterHelpers.hpp]:
```suggestion
//! Creates a monster.
```

File: Includes/CubbyTower/Helpers/MonsterHelpers.hpp [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+//! \param registry A registry that handles entities.
+//! \param health The health of a monster that will be created
+//! \param targetmask A mask that contains the type of a monster
+void CreateMonster(entt::registry& registry, int health, int targetmask);
```
Comment (utilForever) [inline Includes/CubbyTower/Helpers/MonsterHelpers.hpp]:
```suggestion
//! \param health The health of a monster that will be created.
//! \param typeMask A mask that contains the type of a monster.
void CreateMonster(entt::registry& registry, int health, int typeMask);
```

File: Sources/CubbyTower/Helpers/MonsterHelpers.cpp [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+{
+void CreateMonster(entt::registry& registry, int health, int typemask)
+{
+    auto entity = registry.create();
+    registry.emplace<Tag::Enemy>(entity);
+    registry.emplace<TypeMask>(entity, typemask);
+    registry.emplace<Health>(entity, health);
+    registry.emplace<Position>(entity, 0.0, 0.0);
+    registry.emplace<Distance>(entity, 0.0);
+}
```
Comment (utilForever) [inline Sources/CubbyTower/Helpers/MonsterHelpers.cpp]:
```suggestion
void CreateMonster(entt::registry& registry, int health, int typeMask)
{
    auto entity = registry.create();
    registry.emplace<Tag::Enemy>(entity);
    registry.emplace<TypeMask>(entity, typeMask);
    registry.emplace<Health>(entity, health);
    registry.emplace<Position>(entity, 0.0, 0.0);
    registry.emplace<Distance>(entity, 0.0);
}
```

File: Tests/UnitTests/Helpers/MonsterHelpersTests.cpp [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+#include <CubbyTower/Helpers/MonsterHelpers.hpp>
+
+#include <entt/entt.hpp>
+
+using namespace CubbyTower;
+
+TEST_CASE("[MonsterHelpers] - Make several monsters")
+{
+    entt::registry registry;
+    {
```
Comment (utilForever) [inline Tests/UnitTests/Helpers/MonsterHelpersTests.cpp]:
```suggestion
    entt::registry registry;
    
    {
```

File: Tests/UnitTests/Helpers/MonsterHelpersTests.cpp [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+        auto view = registry.view<Tag::Enemy>();
+        CHECK_EQ(view.size(), 0);
```
Comment (utilForever) [inline Tests/UnitTests/Helpers/MonsterHelpersTests.cpp]:
```suggestion
        CHECK(view.empty());
```

File: Tests/UnitTests/Helpers/MonsterHelpersTests.cpp [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+    }
+    CreateMonster(registry, 10, 0b010);
```
Comment (utilForever) [inline Tests/UnitTests/Helpers/MonsterHelpersTests.cpp]:
```suggestion
    }
    
    CreateMonster(registry, 10, 0b010);
```

File: Tests/UnitTests/Helpers/MonsterHelpersTests.cpp [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+
+    {
+        auto view = registry.view<Tag::Enemy>();
+        CHECK_EQ(view.size(), 1);
+        // multi-type view do not know the number of items
```
Comment (utilForever) [inline Tests/UnitTests/Helpers/MonsterHelpersTests.cpp]: What is this? 🤔

File: Tests/UnitTests/Helpers/MonsterHelpersTests.cpp [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+    }
+    {
```
Comment (utilForever) [inline Tests/UnitTests/Helpers/MonsterHelpersTests.cpp]:
```suggestion
    }
    
    {
```

**Review by utilForever — CHANGES_REQUESTED** [review]
You should change some code. Please check them. 🦅

File: Tests/UnitTests/Helpers/MonsterHelpersTests.cpp (hunk 8 above)
Comment (utilForever) [reply Tests/UnitTests/Helpers/MonsterHelpersTests.cpp]:
```suggestion
        // NOTE: Multi-view don't know the number of items
```

File: Sources/CubbyTower/Helpers/MonsterHelpers.cpp (hunk 1 above)
Comment (IHIHHIHI) [inline Sources/CubbyTower/Helpers/MonsterHelpers.cpp]:
```suggestion
```

**Review by utilForever — APPROVED** [review]
I checked all reivews are resolved. LGTM now. 🕊️

**Review by red1108 — APPROVED** [review]

**Comment by red1108** [comment]
LGTM :)

**Pull request merged**